### ISSUE TITLE
[STAC] core dump not related to avx512 

### ISSUE THREAD

**Issue by rkgithubs** [DynamoRIO/dynamorio#3618]
referring to closed #3610 
I ran a different STAC binary (built for broadwell, predecessor to skylake). This binary was built without avx-512 support. This binary crashes as well, i don't think this error has anything to do with avx512 opcode in particular. 

I ran without debug flag. 

/root/rahul/DynamoRIO-x86_64-Linux-7.90.18019-0/bin64/drrun -- /bin/bash run_stac_a2_experiment.sh   Cleaning out old results in "./output/pre_audit_skx_1_socket"... Done. Current directory is: /root/rahul/min_run_scripts  STARTING TESTS at Thu May 16 04:30:15 PDT 2019 ./stac-a2.sh: line 50: 164274 Illegal instruction     (core dumped) ./stac-a2 stac-a2 ret val: 132

error says line 50: 164274 Illegal instruction     (core dumped)
i checked line 50 in the stac-a2.sh shell script, it basically invokes stac-a2 binary. 

47 else
48 #if [["$check_condition"]]; then
49 ./stac-a2
50 fi

I'm also checking the logs with -debug -loglevel 3. There is no log folder on this crashing PID 164504 which makes it difficult to get any more info than what we have on console. 
Please advice how can we provide more information on this crash. 

root@fmx190:~/rahul/DynamoRIO-x86_64-Linux-7.90.18019-0/bin64# ls ../logs/bash.164504          
ls: cannot access '../logs/bash.164504': No such file or directory

<-- execve ./stac-a2 -->
<Starting application /root/rahul/min_run_scripts/stac-a2 (164504)>
<Initial options = -loglevel 3 -code_api -stack_size 56K -signal_stack_size 32K -max_elide_jmp 0 -max_elide_call 0 -early_inject -emulate_brk -no_inline_ignored_syscalls -native_exec_default_list '' -no_native_exec_managed_code -no_indcall2direct >
<Paste into GDB to debug DynamoRIO clients:
set confirm off
add-symbol-file '/root/rahul/DynamoRIO-x86_64-Linux-7.90.18019-0/lib64/debug/libdynamorio.so' 0x00007effd7754580
>
<curiosity: rex.w on OPSZ_6_irex10_short4!>
<(1+x) Handling our fault in a TRY at 0x00007effd799dd7c>
<spurious rep/repne prefix @0x00007effd7511f96 (f2 41 ff e3): >
<Invalid opcode encountered>
<Application /root/rahul/min_run_scripts/stac-a2 (164504).  Internal Error: DynamoRIO debug check failure: /home/travis/build/DynamoRIO/dynamorio/core/translate.c:931 false
(Error occurred @9379 frags)
version 7.90.18019, custom build
-loglevel 3 -code_api -stack_size 56K -signal_stack_size 32K -max_elide_jmp 0 -max_elide_call 0 -early_inject -emulate_brk -no_inline_ignored_syscalls -native_exec_default_list '' -no_native_exec_managed_code -no_indcall2direct 
0x00007effb75a4b40 0x00007effd77ef00e
0x00007effb75a4d90 0x00007effd790568d
0x00007effb75a4e90 0x00007effd790721c
0x00007effb75a5920 0x00007effd790779a
0x00007effb75a5960 0x00007effd78e03ae
0x00007effb75a59c0 0x00007effd79d452f
0x00007effb75a5ec0 0x00007effd79d9c1e
0x00007effb75a61d0 0x00007effd79dcd1e
0x00007effb75a6370 0x00007effd799d89d
0x00007effb5798500 0x0000000000000000>
stac-a2 ret val: 255

**Comment by AssadHashmi** [comment]
There should be a `logs` directory in your main DynamoRIO directory:
```
$ ls -1
ACKNOWLEDGEMENTS
bin64
cmake
ext
include
lib64
License.txt
logs             <--
README
samples
tools
$
```

In your output `drrun` is at:
```
/root/rahul/DynamoRIO-x86_64-Linux-7.90.18019-0/bin64/drrun
```
which implies your logs directory should be at:
```
/root/rahul/DynamoRIO-x86_64-Linux-7.90.18019-0/logs
```

What is the output of `ls -l /root/rahul/DynamoRIO-x86_64-Linux-7.90.18019-0` ?
And `file /root/rahul/DynamoRIO-x86_64-Linux-7.90.18019-0/logs` ?
And `ls -l /root/rahul/DynamoRIO-x86_64-Linux-7.90.18019-0/logs` ?

**Comment by rkgithubs** [comment]
/rahul/min_run_scripts# /root/rahul/DynamoRIO-x86_64-Linux-7.90.18019-0/bin64/drrun -debug -loglevel 3 -- /bin/bash /root/rahul/min_run_scripts/run_stac_a2_experiment.s

stac-a2 PID => 189703 
There is no corresponding log file for the PID which caused crash. 

/rahul/DynamoRIO-x86_64-Linux-7.90.18019-0# ls logs/
bash.189655.00000000  bash.189657.00000000  bash.189659.00000000  bash.189661.00000000  README
bash.189656.00000000  bash.189658.00000000  bash.189660.00000000  bash.189704.00000000

 Please let me know if any more info can be extracted otherwise; line address perhaps so we can objdump the binary and see which instruction is causing it to crash. 
Where can i find this invalid opcode itself we can find out which instruction it is. 

<(1+x) Handling our fault in a TRY at 0x00007fe682301d7c>
<spurious rep/repne prefix @0x00007fe681e75f96 (f2 41 ff e3): >
<Invalid opcode encountered>
<Application /root/rahul/min_run_scripts/stac-a2 (189703).  Internal Error: DynamoRIO debug check failure: /home/travis/build/DynamoRIO/dynamorio/core/translate.c:931 false
(Error occurred @9394 frags)
version 7.90.18019, custom build
-loglevel 3 -code_api -stack_size 56K -signal_stack_size 32K -max_elide_jmp 0 -max_elide_call 0 -early_inject -emulate_brk -no_inline_ignored_syscalls -native_exec_default_list '' -no_native_exec_managed_code -no_indcall2direct

**Comment by rkgithubs** [comment]
<Application /root/rahul/min_run_scripts/stac-a2 (189703).  Internal Error: DynamoRIO debug check failure: /home/travis/build/DynamoRIO/dynamorio/core/translate.c:931 false
(Error occurred @9394 frags)
version 7.90.18019, custom build
-loglevel 3 -code_api -stack_size 56K -signal_stack_size 32K -max_elide_jmp 0 -max_elide_call 0 -early_inject -emulate_brk -no_inline_ignored_syscalls -native_exec_default_list '' -no_native_exec_managed_code -no_indcall2direct 
0x00007fe661f08b40 0x00007fe68215300e
0x00007fe661f08d90 0x00007fe68226968d
0x00007fe661f08e90 0x00007fe68226b21c
0x00007fe661f09920 0x00007fe68226b79a
0x00007fe661f09960 0x00007fe6822443ae
0x00007fe661f099c0 0x00007fe68233852f
0x00007fe661f09ec0 0x00007fe68233dc1e
0x00007fe661f0a1d0 0x00007fe682340e38
0x00007fe661f0a370 0x00007fe68230189d
0x00007fe6600fc500 0x0000000000000000>
stac-a2 ret val: 255

**Comment by rkgithubs** [comment]
I checked with -debug -loglevel 1 opcode is not printed anywhere. 
/root/rahul/DynamoRIO-x86_64-Linux-7.90.18019-0/bin64/drrun -debug -loglevel 1 -- /bin/bash /root/rahul/min_run_scripts/run_stac_a2_experiment.sh

<Paste into GDB to debug DynamoRIO clients:
set confirm off
add-symbol-file '/root/rahul/DynamoRIO-x86_64-Linux-7.90.18019-0/lib64/debug/libdynamorio.so' 0x00007fdb6ab1c580
>
<(1+x) Handling our fault in a TRY at 0x00007fdb6ad65d7c>
<spurious rep/repne prefix @0x00007fdb6a8d9f96 (f2 41 ff e3): >
<Invalid opcode encountered>
<Application /root/rahul/min_run_scripts/stac-a2 (189762).  Internal Error: DynamoRIO debug check failure: /home/travis/build/DynamoRIO/dynamorio/core/translate.c:931 false
(Error occurred @9398 frags)
version 7.90.18019, custom build
-loglevel 1 -code_api -stack_size 56K -signal_stack_size 32K -max_elide_jmp 0 -max_elide_call 0 -early_inject -emulate_brk -no_inline_ignored_syscalls -native_exec_default_list '' -no_native_exec_managed_code -no_indcall2direct 
0x00007fdb4a96cb40 0x00007fdb6abb700e
0x00007fdb4a96cd90 0x00007fdb6accd68d
0x00007fdb4a96ce90 0x00007fdb6accf21c
0x00007fdb4a96d920 0x00007fdb6accf79a
0x00007fdb4a96d960 0x00007fdb6aca83ae
0x00007fdb4a96d9c0 0x00007fdb6ad9c52f
0x00007fdb4a96dec0 0x00007fdb6ada1c1e
0x00007fdb4a96e1d0 0x00007fdb6ada4e38
0x00007fdb4a96e370 0x00007fdb6ad6589d
0x00007fdb48b60500 0x0000000000000000>
stac-a2 ret val: 255

**Comment by rkgithubs** [comment]
these are in-house built binaries. 

spurious rep/repne prefix @0x00007fdb6a8d9f96 (f2 41 ff e3): 
Invalid opcode encountered
Application /root/rahul/min_run_scripts/stac-a2 (189762).  Internal Error: DynamoRIO debug check failure: /home/travis/build/DynamoRIO/dynamorio/core/translate.c:931 false

**Comment by derekbruening** [comment]
As noted earlier, the 'spurious rep/repne prefix' for `f2 41...` is spurious: please ignore it (it is already eliminated in the latest DR code: #1978).

The `Invalid opcode encountered` is the key.  That phrase will be printed to a log file: search all the logs for it and look at the subsequent lines.  Or go into the DR code (`core/arch/x86/decode.c` as pointed at in one of the other issues) and `s/LOG(THREAD_GET, LOG_ALL, 1, /SYSLOG_INTERNAL_WARNING/` which will print it to stderr.

**Comment by rkgithubs** [comment]
STAC crashed on these byte codes. 
SYSLOG_WARNING: Invalid opcode encountered
Error decoding 0x000000000259eb28 == 0x62 0xf2 0xfd 0x48 0x19 0xe
Error decoding 0x000000000259ec43 == 0x62 0xf1 0x7c 0x48 0x28 0xc1
Error decoding 0x000000000259ec43 == 0x62 0xf1 0x7c 0x48 0x28 0xc1
Error decoding 0x000000000259ec43 == 0x62 0xf1 0x7c 0x48 0x28 0xc1
ERROR: recreate_app : looking for 0x0000000040274ee3 in frag @ 0x0000000040274ed4 (tag 0x000000000259ec43)
TAG  0x0000000000000000
 +0    L4 @0x00007fdc615d2930  e9 3e cf 00 a1       jmp    $0x000000000259ec43
END 0x0000000000000000

**Comment by derekbruening** [comment]
AVX-512 again: `62f2fd48190e   vbroadcastsd zmm1 , qword ptr [rsi]`

We should have put a simple check for evex prefixes with a fatal message and exit long ago as @hgreving2304 is suggesting now (though back then we didn't know how long it would take to get around to adding AVX-512 support).

**Comment by rkgithubs** [comment]
Yes, we saw the same after disassembling the binary- too many evex instructions. Code came from another static library that was compiled with avx512. After disabling it stac-a2 is able to run successfully via DR. 
There are a few performance issues. Workload maxes out all cores, but after launching via DR (using drcachesim offline for 500M instructions worth of trace collection)- we saw that Core Utilization dropped to ~20%. We need to limit trace collection to fewer number of cores.  I tried -cores option that comes with drcachesim tool but trace is still captured across all cores. This is a 2 socket 28C skylake server configuration. There are 56 trace files in compressed format.
